plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ServiceResourceMapper.java

```diff
@@ -0,0 +1,163 @@
+		return resource.getServiceName().equals(s.getName());
+	}
+
+	private static boolean isRelated(IService s, IDeploymentConfig dc) {
+		return containsAll(s.getSelector(), dc.getReplicaSelector());
+	}
+
+	private static boolean isRelated(IService s, IPod pod) {
+		return containsAll(s.getSelector(), pod.getLabels());
+	}
+
+	private static Collection<IResource> getRelated(Collection<IResource> resources, IPod resource) {
+		return getRelatedReplicationControllers(resources, Collections.singleton(resource));
+	}
+
+	private static Collection<IResource> getRelated(Collection<IResource> resources, IDeploymentConfig dc) {
+		Collection<IResource> result = new HashSet<>();
+
+		Collection<String> dcImageRefs = computeImageRefs(dc);
+
+		result.addAll(getRelatedImageTags(resources, dcImageRefs, dc));
+		Collection<IBuildConfig> buildConfigs = getRelatedBuildConfigs(resources, dcImageRefs, dc);
+		result.addAll(buildConfigs);
+		Collection<IBuild> builds = getRelatedBuilds(resources, dcImageRefs, buildConfigs);
+		result.addAll(builds);
+		Collection<IPod> pods = getRelatedPods(resources, builds, dc);
+		result.addAll(pods);
+		result.addAll(getRelatedReplicationControllers(resources, pods));
+		return result;
+	}
+
+	private static Collection<IResource> getRelatedReplicationControllers(Collection<IResource> resources,
+			Collection<IPod> pods) {
+		Collection<String> deploymentNames = pods.stream()
+				.filter(r -> r.isAnnotatedWith(OpenShiftAPIAnnotations.DEPLOYMENT_NAME))
+				.map(r -> r.getAnnotation(OpenShiftAPIAnnotations.DEPLOYMENT_NAME)).collect(Collectors.toSet());
+
+		Collection<IResource> result = new HashSet<>();
+		resources.forEach(r -> {
+			if (ResourceKind.REPLICATION_CONTROLLER.equals(r.getKind()) && deploymentNames.contains(r.getName())) {
+				result.add(r);
+			}
+		});
+		return result;
+	}
+
+	private static Collection<IPod> getRelatedPods(Collection<IResource> resources, Collection<IBuild> builds,
+			IDeploymentConfig dc) {
+
+		Collection<String> buildNames = builds.stream().map(bc -> bc.getName()).collect(Collectors.toSet());
+
+		return resources.stream().filter(r -> {
+			return r instanceof IPod && (buildNames.contains(r.getAnnotation(OpenShiftAPIAnnotations.BUILD_NAME))
+					|| dc.getName().equals(r.getAnnotation(OpenShiftAPIAnnotations.DEPLOYMENT_CONFIG_NAME)));
+		}).map(r -> (IPod) r).collect(Collectors.toSet());
+
+	}
+
+	private static Collection<IBuild> getRelatedBuilds(Collection<IResource> resources, Collection<String> dcImageRefs,
+			Collection<IBuildConfig> buildConfigs) {
+
+		Collection<IBuild> result = new HashSet<>();
+		Collection<String> bcNames = buildConfigs.stream().map(bc -> bc.getName()).collect(Collectors.toSet());
+		resources.forEach(r -> {
+			if (r instanceof IBuild) {
+				IBuild build = (IBuild) r;
+				if (bcNames.contains(r.getLabels().get(OpenShiftAPIAnnotations.BUILD_CONFIG_NAME)) || dcImageRefs.contains(imageRef(build))) {

```

Comment:
@jcantrill , I found that StartBuildHandler.loadBuildConfig(build) takes the same way to get build config name from build, but in Deployment it was always received as an annotation value. Are the two ways equivalent?